src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -77,18 +77,23 @@ def select_interpreter_for_targets(self, targets):
     """Pick an interpreter compatible with all the specified targets."""
     tgts_by_compatibilities = defaultdict(list)
     filters = set()
-    for target in targets:
-      if isinstance(target, PythonTarget):
-        c = self._python_setup.compatibility_or_constraints(target)
-        tgts_by_compatibilities[c].append(target)
-        filters.update(c)
+
+    if self._python_setup.get_options().is_flagged('interpreter_constraints'):
```

Comment:
Mentioned offline: this separate if/else block should not actually be necessary if the `is_flagged` logic were to move into `compatibility_or_constraints`... the effect when `is_flagged=True` would be that all targets ended up with the same config values.

This would also have the advantage of working anywhere else the `compatibility_or_constraints` method is used.